core/changedetector/utils.go

```diff
@@ -0,0 +1,94 @@
+package changedetector
+
+import (
+	"github.com/golang/protobuf/protoc-gen-go/descriptor"
+	"google.golang.org/protobuf/proto"
+)
+
+func isMessageFieldChanged(field1, field2 *descriptor.FieldDescriptorProto) bool {
```

Comment:
We don't have unit test cases for other utils too. Also, utils test cases are covered inside proto_change_detector_service_test as I have not mocked any helper methods and covered all scenarios.

Reply:
1. 
I don't see any "deprecated field" in the testing part of proto_change_detector_service_test, additionally it is hard to correlate these methods with tests in proto_change_detector_service_test.

2.
Just because you haven't mocked any of these methods doesn't mean we shouldn't test them. We should have unit tests for this, as it can be a solid standalone piece with its own unit tests.

3.
Say anyone in the future who is adding function to this file, will most likely not write unit test case, as this becomes new agreed convention, but may also miss to add all the valid scenarios in the integrated place.

Reply:
updated as per our discussion